gradoop-store/gradoop-store-api/src/main/java/org/gradoop/common/storage/predicate/filter/impl/PropReg.java

```diff
@@ -0,0 +1,67 @@
+/**
+ * Copyright © 2014 - 2018 Leipzig University (Database Research Group)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.gradoop.common.storage.predicate.filter.impl;
+
+import org.gradoop.common.storage.predicate.filter.api.ElementFilter;
+
+import javax.annotation.Nonnull;
+import java.util.regex.Pattern;
+
+/**
+ * property filter by regex
+ *
+ * @param <FilterImpl> filter implement type
+ */
+public abstract class PropReg<FilterImpl extends ElementFilter>
+  implements ElementFilter<FilterImpl> {
+
+  /**
+   * regex pattern
+   */
+  private final Pattern reg;
+
+  /**
+   * property key name
+   */
+  private final String key;
+
+  /**
+   * label regex filter constructor
+   *
+   * @param key property key
+   * @param reg label regex
+   */
+  public PropReg(
+    @Nonnull String key,
+    @Nonnull Pattern reg
+  ) {
+    this.key = key;
+    this.reg = reg;
+  }
+
+  @Override
+  public String toString() {
+    return String.format("e.prop.%1$s SIMILAR TO `%2$s`", key, reg.pattern());
```

Comment:
Same. `SIMILAR TO` fits not in my opinion. More like `MATCHES PATTERN` or `MATCHES REGEX`